### DISCUSSION TITLE
I'm overwhelmed and unsure where to start

### DISCUSSION THREAD

**Discussion by FarelTheGiraffe** [community/community#22897] [New to GitHub]
<p>Hello, I’ve had several years since I last programmed anything. Now I’m getting back into it, and I’m doing that by designing a web application that simulates the results of a certain game, based on player input.</p>
<p>Of course I heard a lot about GitHub being great for projects like these, and supposedfly when people apply for programming jobs (which I likely will do in the future) they just link to their GitHub page, and that’s why I opened this up, however as many videos as I watch, I cannot get a grasp on how this is meant to work. </p>
<p>-I created a repository, but I do not know how to put files onto it</p>
<p>-I do not know if they’d at all function on this site, or if it’s just static code</p>
<p>-I don’t know if GitHub provides its own SQL database</p>
<p>-I do not understand why a repository I created online in the web-browser does not appear in the desktop version of GitHub</p>
<p>-Shouldn’t I publish my webapllication on my private website? And if I did, what would be left here?</p>
<p>These are likely some initial beginner problems, and I apologize if they’re obvious, but I cannot get started with anything yet, and I already lost a few days trying to get behind it. I apologize, but I just have to ask.</p>
<p>Thank you for your time</p>

**Comment by seveas** [comment, marked as answer]
<p>Hi <a class="mention" href="/farelthegecko">@farelthegecko</a>,</p>
<p>Welcome to GitHub, and welcome back to programming!</p>
<p>I’ll try to clear up some of the confusion you are experiencing, git and github can indeed be overwhelming if you’re starting afresh.</p>
<p>For your first and fourth question: after creating a repository on <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a>, you have to clone it (using git, or the GitHub desktop) to your computer to add files to it and make further changes. You can then push these changes back up to GitHub to share them with others.</p>
<p>And about hosting: neither git, nor GitHub are a way of serving applications. Git is a source code management tool, and GitHub is a place where you can store your repositories and collaborate with others on your source code. So for your second, third and fifth question: GitHub is there for looking at the source code, there’s no execution environment and no databases for you to use, so you’ll indeed have to use your own website/server to host the actual running application.</p>
<p>There’s one exception to this: GitHub also has a product called GitHub pages, where GitHub will host a completely static website for you. But that’s limited to static websites or jekyll-based static generated websites only.</p>
<p>Hopefully that cleared a few things up for you! If you have other questions, don’t hesitate to ask.</p>

**Comment by angela-crist** [comment]
<p>Hi <a class="mention" href="/farelthegecko">@farelthegecko</a>,</p>
<p>Thanks for being part of the GitHub Community Forum! We’re glad you’re here and welcome to GitHub.</p>
<p>Along with <a class="mention" href="/seveas">@seveas</a> great insight in response to your questions, you may want to check out the <a href="https://lab.github.com/" rel="nofollow noopener">GitHub Learning Lab</a> for some hands-on guidance. The Learning Lab bot guides you through projects and provides feedback right from your GitHub repository. If you find yourself stuck in a course from the Learning Lab you can post questions in <a href="/t5/GitHub-Learning-Lab/bd-p/learn">the board specific to the Learning Lab</a>. </p>
<p>Happy coding!</p>
